Messages

alshakerM opened the pull request:
Added messages and buttons to the icons on the navBar to access said messages.
Also changed the icons color when they are clicked.

alshakerM commented:
Fixes a bug that happens when you click on one of the users that makes the icon color changes back to white.

alshakerM commented:
Deleted unused imports

alshakero commented on src/Messages/Messages.css:
```diff
@@ -0,0 +1,26 @@
+  max-width: 236px;
+  min-height: 44px;
+  padding: 16px;
+  justify-self: center;
+  align-self: center;
+  text-align: center;
```
These are unneeded since it's not a flex.

alshakero commented on src/Messages/Messages.css:
```diff
@@ -0,0 +1,26 @@
+}
+
+.messageText.is-sent {
+  background: #efefef;
+  display: flex;
+  justify-content: flex-end;
+  margin-right: 20px;
```
You only need the background color and the margin.

alshakero commented on src/Messages/Messages.css:
```diff
@@ -0,0 +1,26 @@
+  display: flex;
+  justify-content: flex-end;
+  margin-right: 20px;
+}
+.messages-user-thumbnail-message-text {
+  display: flex;
+  align-items: center;
+  gap: 10px;
+}
+.messages-user-thumbnail-message-text.is-sent {
+  align-self: flex-end;
+}
```
You can add this file to the CSS module.

alshakero commented on src/Messages/Messages.js:
```diff
@@ -0,0 +1,270 @@
+    <div className={userSection}>
+      <div className={usernameIconSection}>
+        <p>
+          <strong>7kami_me</strong>{' '}
```
```suggestion
          <strong>7kami_me</strong>
```

alshakero commented on src/Messages/Messages.js:
```diff
@@ -0,0 +1,270 @@
+          </Link>
+          <div>
+            <p className={senderUsername}>{user.from_username}</p>
+            <p className={hasSentMessage}>Active 20min ago</p>
```
Why lock it to 20 mins ago? It should be the relative time of the last message.

alshakero commented on src/Messages/Messages.js:
```diff
@@ -0,0 +1,270 @@
+      messagesData[userIndex]?.messages.length - 1
+    ].sent_on
+  );
+  console.log(fromUserId);
```
```suggestion

```

alshakero commented on src/Messages/Messages.module.css:
```diff
@@ -0,0 +1,191 @@
+.messagesPageContainer {
+  max-width: 935px;
+  margin: 80px auto;
+  display: flex;
+}
```
This needs a border radius. Check IG for reference.

alshakero commented on src/Messages/Messages.module.css:
```diff
@@ -0,0 +1,191 @@
+}
+.usernameIconSection {
+  display: flex;
+  margin: 30px 20px 30px 120px;
```
I appreciate the creativity, but this not good. Writing the coordinates manually makes your CSS brittle (it breaks easily). I would add an empty `div` in `userSection`. The give `flex: 1` to everything inside `userSection`. This way you'll have 3 equal parts with the username in the center. 

```css
userSection > * {
	flex: 1
}
```

alshakero commented on src/Messages/Messages.module.css:
```diff
@@ -0,0 +1,191 @@
+  width: 350px;
+  border: 1px solid #dbdbdb;
+  overflow: hidden auto;
+  height: 50%;
```
hmmmm. 50% filling the viewport means the container (`leftSection`) is too big. That's why you have empty scrolling below on the screen. Make this 100%, then make the container smaller. I'd make the container of this div like this

```css
.leftSection {
	/* screen height - margin top - margin bottom */
	height: calc( 100vh - 80px - 80px); 
}
```

If you do this for both sides (left and right) you'll get the same responsiveness as IG.

alshakerM replied:
On it

alshakero commented on src/Messages/Messages.js:
```diff
@@ -0,0 +1,270 @@
+  sendAMessage,
+  senderInfo,
+  sendersSection,
+  SenderThumbnail,
```
```suggestion
  senderThumbnail,
```

alshakero commented on src/Messages/Messages.module.css:
```diff
@@ -0,0 +1,191 @@
+  gap: 10px;
+  align-items: center;
+}
+.SenderThumbnail {
```
```suggestion
.senderThumbnail {
```

alshakero commented on src/Messages/Messages.module.css:
```diff
@@ -0,0 +1,191 @@
+  padding: 20px;
+}
+.messagesInput {
+  height: 18px !important;
```
I don't think you need this.
```suggestion
  height: 18px;
```

alshakero commented on src/Messages/Messages.module.css:
```diff
@@ -0,0 +1,191 @@
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  border-radius: 50%;
```
The icon isn't centered inside this. Maybe some padding-top of 10px and box-sizing obviously.

alshakero reviewed (changes requested):
Loving it! Left some comments. I'll leave some here as well:

1. The message entry should have a background when it's active. See here 
[here](https://user-images.githubusercontent.com/17054134/143139204-cf2376e5-8e13-4eaa-882f-54613259346e.png).
2. The whole message entry should be clickable, not only the photo.

alshakerM commented on src/Icons/Icons.js:
```diff
@@ -1,33 +1,48 @@
+          viewBox="0 0 48 48"
+          width="22"
+        >
+          {location.pathname === '/direct/inbox' || location.pathname === `/direct/t/${from_user_id}`? (
```
Didn't know there was a startsWith 🗡️ thanks for the info

alshakerM commented:
the title says it all 👯

alshakerM commented:
Changed the if statement in icons to startsWith instead of || for cleaner and DRYer code.

alshakerM commented:
Deleted some unused CSS and edit the height of the senderSection and messagesSection.

alshakerM commented:
Added some padding to the main telegram icon and made it a bit larger to look like the real IG.
Edited a className to be camelCase.
Removed !important from messagesInput.

alshakerM commented:
Deleted the CSS file and moved its content to the module CSS file.
Remove height from the messagesSection.
Merged Routes in App.js.

alshakerM commented:
Changed the time in senderSection.
Removed console log.
Added border radius ( not the main div but to each corner section to avoid using overflow ).
Removed margin from the icon in userSection and added an empty div.

alshakero commented on src/Messages/Messages.js:
```diff
@@ -0,0 +1,285 @@
+import { Link } from 'react-router-dom';
+import { NavBar } from '../NavBar/NavBar';
+import { absoluteToRelativeDate } from '../utils';
+import {
+  arrowDownIcon,
+  emptyImg,
+  hasSentMessage,
+  isSent,
+  lastMessageSentDate,
+  leftSection,
+  messagesBody,
+  messagesInput,
+  messagesInputContent,
+  messagesInputSection,
+  messagesPageContainer,
+  messagesSection,
+  messagesSenderAvatar,
+  messagesSenderInfo as messagesHeader,
+  messagesSenderUserInfo,
+  messagesSenderUsername,
+  messagesUserThumbnailMessageText,
+  messageText,
+  rightSection,
+  sendAMessage,
+  senderInfo,
+  sendersSection,
+  senderThumbnail,
+  senderUsername,
+  sendMessageButton,
+  sendMessageIcon,
+  sendMessagesSection,
+  telegramIcon,
+  telegramIconContainer,
+  usernameIconSection,
+  userSection,
+  whenWasActive,
+  yourMessages,
+  emptyUserName,
+} from './Messages.module.css';
```
In the future you can import all these like so:

```js
import styles from './Messages.module.css'
```

then 
```jsx
<div className={styles.messageText} ...
```

alshakero commented on src/Messages/Messages.js:
```diff
@@ -0,0 +1,285 @@
+      {messagesData.map((user) => (
+        <Link
+          to={`/direct/t/${user.from_user_id}`}
+          style={
+            user.from_user_id === fromUserId
+              ? { backgroundColor: '#efefef' }
+              : {}
+          }
+          className={senderInfo}
```
We can let this slip. But it's better to create `isSelected` class and use classnames.

alshakero commented on src/Messages/Messages.js:
```diff
@@ -0,0 +1,285 @@
+              </div>
+              <svg
+                aria-label="View thread details"
+                class="_8-yf5 "
```
Unneeded class.

alshakero commented on src/Messages/Messages.js:
```diff
@@ -0,0 +1,285 @@
+                <div className={messagesInputContent}>
+                  <svg
+                    aria-label="Emoji"
+                    class="_8-yf5 "
```
Unneeded class.

alshakero commented on src/Messages/Messages.js:
```diff
@@ -0,0 +1,285 @@
+                  />
+                  <svg
+                    aria-label="Add Photo or Video"
+                    class="_8-yf5 "
```
+1

alshakero commented on src/NavBar/NavBar.js:
```diff
@@ -97,7 +97,7 @@ export function NavBar() {
         </div>
         <div className="nav-icons">
-          <Icons />
+          <Icons from_user_id={from_user_id}/>
```
```suggestion
          <Icons fromUserId={from_user_id}/>
```

alshakero reviewed (approved):
Looks amazing. Left some cosmetic remarks. Feel free to merge after addressing them.

alshakero commented:
"Final changes" isn't a great commit message, given you merged it to master. Please either squash your PRs or pay attention to commit messages.
